Make htmllint a peerDependency

sventschui opened the pull request:
Make htmllint a peerDependency to be able to upgrade htmllint without upgrading gulp-htmllint

yvanavermaet commented:
@sventschui Could you elaborate why you'd change htmllint into a peerDependency? I'm not very familiar with peer dependencies if I'm totally honest.

If I understand correctly the peerDependencies will be installed at the top-level to avoid modules installing multiple versions of that peer dependency.

In our case, I don't think that's an issue. htmllint is very specific and I doubt there is any other module that uses it.
+
In npm3 peer dependencies won't be automatically installed, which means that everyone who uses gulp-htmllint will need to manually configure htmllint in their package.json

An interesting discussion: https://github.com/npm/npm/issues/6565

Are there things I'm overseeing/missing?

sventschui commented:
We had the case of having htmllint as a dependency twice. Since we refactored our build and dropped the second use this PR may be obsolete. Feel free to either merge or close

yvanavermaet commented:
As stated in my previous comment, I'm not convinced that htmllint should be a peerDependency. Will close this Pull Request. Thanks though :-)
